### PR TITLE
Refactors ship class into parent with subclasses inheriting it for each type.  Adds tests.

### PR DESCRIPTION
**Pull request by xMestas**
Improves line coverage to 96% in modules package.

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Battleship.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package cs361.battleships.models;
+
+import java.util.ArrayList;
+
+public class Battleship extends Ship {
+
+    public Battleship() {
+        this.occupiedSquares= new ArrayList<Square>();// constructs list
+        this.kind="BATTLESHIP";// sets kind equal to that passed in
+        this.occupiedSquares.add(new Square(0, 'A'));
```
Comment (chase-stubbs) [inline src/main/java/cs361/battleships/models/Battleship.java]: My guess is that what this does is it sets four occupied locations for battleship and then re-sets them later? Anyway, this function doesn't initially explain itself, so maybe a comment?

File: src/main/java/cs361/battleships/models/Ship.java [hunk 2]
```diff
@@ -7,36 +7,13 @@
...
-	private int length;// holds the number of spaces the ship takes up
+	@JsonProperty protected List<Square> occupiedSquares;// holds the locations on the board the is present at
+	protected String kind; // holds the kind of ship that is present
+	protected int length;// holds the number of spaces the ship takes up
 
 	public Ship() {
 		occupiedSquares = new ArrayList<>();
 	}
-	
-	public Ship(String kind) {
```
Comment (chase-stubbs) [inline src/main/java/cs361/battleships/models/Ship.java]: good work getting rid of this unpleasant-to-look-at function

File: src/main/java/cs361/battleships/models/Battleship.java (hunk 1 above)
Comment (xMestas) [reply src/main/java/cs361/battleships/models/Battleship.java]: Just the constructor setting the default values.

**Comment by xMestas** [comment]
Resolves #28 and #29

**Pull request merged**